src/common/tvgMath.cpp

```diff
@@ -120,9 +120,15 @@ bool inverse(const Matrix* m, Matrix* out)
                m->e12 * (m->e21 * m->e33 - m->e23 * m->e31) +
                m->e13 * (m->e21 * m->e32 - m->e22 * m->e31);
 
-    if (tvg::zero(det)) return false;
+    auto scaled = false;
 
-    auto invDet = 1 / det;
```

Comment:
how about just checking this?
```if (std::isnan(invDet)) return false;```

Reply:
in order to handle nans and infs I used isfinite.

Reply:
hmm... inf is necessary to check here?